node/src/node_configurator/mod.rs

```diff
@@ -111,7 +111,10 @@ pub fn data_directory_from_context(
     chain: Chain,
 ) -> PathBuf {
     match data_directory_opt {
-        Some(data_directory) => data_directory.clone(),
+        Some(data_directory) => {
+            let data_dir = data_directory.clone().to_string_lossy().to_string();
```

Comment:
I've got a feeling that you might be able to get rid of this clone() here. My reasoning is that you usually don't need to have it owned if the (displayable) look of it is the only thing you want. Here you just convert it to a string.